File: 33/cpp/main.cc
#include <ctime>
#include <iostream>
#include <vector>

#include "user_input.h"

namespace {

int generate_ramdom_index(int max) {
    static bool seed = false;
    if (!seed) {
        std::srand(std::time(nullptr));
        seed = true;
    }
    return std::rand() % max;
}

}  // namespace

int main() {
    std::vector<std::string> answers = {"Yes", "No", "Maybe",
                                        "Ask again later"};
    read_user_input("What's your question? ");

    std::cout << answers[generate_ramdom_index(answers.size())] << "."
              << std::endl;

    return 0;
}

File: 33/go/main.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func generateRamdomIndex(max int) int {
	r := rand.New(rand.NewSource(time.Now().UnixNano()))
	return r.Int() % max
}

func main() {

	answers := []string{"Yes", "No", "Maybe", "Ask again later"}
	readUserInput("What's your question? ")
	answer := answers[generateRamdomIndex(len(answers))]
	fmt.Printf("%s.\n", answer)
}
